server/grammar/YmlParser.g4

```diff
@@ -294,10 +171,10 @@ nonArithmeticValue: chainedCall | bool | STRING | DATE;
 
 instanciationVariable: QUESTION_MARK ymlId;
 
+expressionWithIndex: expression withIndex*;
```

Comment:
Huuuum… Confusing “with” in rule name since there can be no index here (`*`). Also, the rule name `withIndex` sounds strange to me too since it’s _just_ an index and we don’t know _what_ is “with” it.

```
possiblyIndexedExpression: expression index*;
```

? Maybe you’ll say it’s ugly as hell.